File: SIN BDD/RESTFULL_JAVA_CONUNI_GR06/08. SERVIDOR/servidor_restful_java_conuni_gr06/src/java/ec/edu/monster/servicio/ServicioLongitud.java
package ec.edu.monster.servicio;

public class ServicioLongitud {

    public double metrosAPies(double metros) {
        return metros * 3.28084;
    }

    public double kilometrosAMillas(double kilometros) {
        return kilometros * 0.621371;
    }

    public double centimetrosAPulgadas(double centimetros) {
        return centimetros / 2.54;
    }

    public double yardasAMetros(double yardas) {
        return yardas / 1.09361;
    }

    public double milimetrosAPulgadas(double milimetros) {
        return milimetros * 0.0393701;
    }
}

File: SIN BDD/SOAP_DOTNET_CONUNI_GR06/07. CLIMOVIL/cliente_movil_soap_dotnet_conuni_gr06/src/Servicio/ServicioLongitud.cs
namespace Ec.Edu.Monster.Servicio;

public class ServicioLongitud
{
    private readonly Utilidades.ClienteSoap cliente = new();

    public double MetrosAPies(double metros) => cliente.Ejecutar(servicio => servicio.MetrosAPies(metros));

    public double KilometrosAMillas(double kilometros) => cliente.Ejecutar(servicio => servicio.KilometrosAMillas(kilometros));

    public double CentimetrosAPulgadas(double centimetros) => cliente.Ejecutar(servicio => servicio.CentimetrosAPulgadas(centimetros));

    public double YardasAMetros(double yardas) => cliente.Ejecutar(servicio => servicio.YardasAMetros(yardas));

    public double MilimetrosAPulgadas(double milimetros) => cliente.Ejecutar(servicio => servicio.MilimetrosAPulgadas(milimetros));
}
